lib/charms/prometheus_k8s/v0/prometheus.py

```diff
@@ -573,48 +579,91 @@ def _labeled_static_job_config(self, job, job_name_prefix, hosts, scrape_metadat
-        config["static_configs"] = []
+        for static_config in original_static_configs:
+            labels = static_config.get("labels", {})
+            targets = static_config.get("targets", [])
 
-        relabel_config = {
-            "source_labels": ["juju_model", "juju_model_uuid", "juju_application"],
-            "separator": "_",
-            "target_label": "instance",
-            "regex": "(.*)",
-        }
-
-        for static_config in static_configs:
-            labels = static_config.get("labels", {}) if static_configs else {}
-            all_targets = static_config.get("targets", [])
-
-            ports = []
-            unitless_targets = []
-            for target in all_targets:
-                host, port = target.split(":")
-                if host.strip() == "*":
-                    ports.append(port.strip())
-                else:
-                    unitless_targets.append(target)
-
-            if unitless_targets:
-                unitless_config = self._labeled_unitless_config(
-                    unitless_targets, labels, scrape_metadata
+            if unitless_targets := [
+                target for target in targets if not target.startswith("*:")
+            ]:
+                static_configs.append(
+                    self._labeled_unitless_config(
+                        unitless_targets, labels, scrape_metadata
+                    )
                 )
-                config["static_configs"].append(unitless_config)
 
-            for host_name, host_address in hosts.items():
-                static_config = self._labeled_unit_config(
-                    host_name, host_address, ports, labels, scrape_metadata
-                )
-                config["static_configs"].append(static_config)
-                if "juju_unit" not in relabel_config["source_labels"]:
-                    relabel_config["source_labels"].append("juju_unit")
+            if ports := [
+                target.split(":")[1] for target in targets if target.startswith("*:")
+            ]:
+                if self._is_aggregated_static_config(static_config, scrape_metadata):
+                    # Do not add targets for proxy applications that publish scrape
+                    # jobs with Juju labels that do not match the scrape metadata
+                    logger.error(
+                        "Received a static_config that uses the '*:<port>' target "
+                        "schema, and that has Juju topology labels that do not match "
+                        "the scrape_metadata"
+                    )
+                    continue
 
-        config["relabel_configs"] = [relabel_config]
+                for host_name, host_address in hosts.items():
+                    static_configs.append(
+                        self._labeled_unit_config(
+                            host_name, host_address, ports, labels, scrape_metadata
+                        )
+                    )
 
-        return config
```

Comment:
This is a rather big return statement, not to mention it makes the function itself rather long. Besides most of the dictionary is boiler plate with only two substitutions - `job["metrics_path"]` and `static_configs` that is easy to miss in the "boiler plate forest". What is being substituted can be made explicit in the context of a separate function for example

```
return self._job_configs(job["metrics_path"], static_configs)
```